compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -62,9 +62,9 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
     super.transform(ref) match
       case ref1: SymDenotation if ref1.is(Method) =>
```

Comment:
Originally the `ref1 ne ref` was here which I think is equivalent to what the code is doing now?

Reply:
Not quite, because now `transformVarArgs` is always called. If I move `ref1 ne ref` back to the `case` guard, some situations are not covered and e.g. `tests/neg/varargs-annot` fails.